File: 1652-defuse-the-bomb/1652-defuse-the-bomb.cpp
class Solution {
public:
    vector<int> decrypt(vector<int>& code, int k) {
        int n = code.size();
        vector <int> ans(n, 0);
        for(int i = 0; i < n; i++) {
            int s = 0;
            for(int j = k; j != 0; j = j < 0 ? j + 1 : j - 1) {
                s += code[(i + j + n) % n];
            }
            ans[i] = s;
        }
        
        return ans;
    }
};

File: 1652-defuse-the-bomb/1652-defuse-the-bomb.js
/**
 * @param {number[]} code
 * @param {number} k
 * @return {number[]}
 */
var decrypt = function(code, k) {
    let out = []
    let n = code.length;
    for(let i = 0; i < n; i++){      
        let sum = 0;
        for(let j = k; j != 0; ){       
            sum += code[(i + j + n) % n];
            if(j < 0) j ++;
            else j--;
        }
        out.push(sum);
    }
    return out; 
};